tests/vxlan/test_vxlan_ecmp.py

```diff
@@ -195,11 +201,11 @@ def get_ethernet_to_neighbors(neighbor_type, minigraph_data):
     '''
 
     pattern = re.compile("{}$".format(neighbor_type))
-    ret_list = []
+    ret_list = {}
 
     for intf in minigraph_data['minigraph_neighbors']:
         if pattern.search(minigraph_data['minigraph_neighbors'][intf]['name']):
-            ret_list.append(intf)
+            ret_list[intf] = minigraph_data['minigraph_neighbors'][intf]['address']
 
     return ret_list
```

Comment:
Why is this change necessary?
We got an error: `KeyError: 'address'`, because the 
minigraph_data['minigraph_neighbors'][intf] looks like 
`"minigraph_neighbors": {
    "Ethernet0": {
      "namespace": "",
      "name": "ARISTA01T2",
      "port": "Ethernet1"
    },
    "Ether...`
and there is no "address" key.
Can we revert this?
Sorry, this error hasn't appeared previously, because on T1-LAG this function doesn't invoke, only on T1.

Reply:
Hi @ihorchekh : I added this change since I needed the mapping of interface to address, not just the interface list. I tested on T1-lag, and it was working. So you are saying it fails on T1. Let me take a look. Do you mind giving me the log file ?

Reply:
I don't have a log file because the test has failed on the setup step, and none of the tests has been run. Here's the traceback:
`19:38:52 __init__._fixture_generator_decorator    L0078 ERROR  | 
KeyError('address',)
Traceback (most recent call last):
  File "/root/mars/workspace/sonic-mgmt/tests/common/plugins/log_section_start/__init__.py", line 74, in _fixture_generator_decorator
    res = next(it)
  File "/root/mars/workspace/sonic-mgmt/tests/vxlan/test_vxlan_ecmp.py", line 613, in setUp
    encap_type_data['neighbor_config'] = configure_vnet_neighbors(data['duthost'], encap_type_data['intf_to_ip_map'], minigraph_data=minigraph_facts, af=payload_version)
  File "/usr/local/lib/python2.7/dist-packages/allure_commons/_allure.py", line 169, in impl
    return func(*a, **kw)
  File "/root/mars/workspace/sonic-mgmt/tests/vxlan/test_vxlan_ecmp.py", line 475, in get_t2_ports
    
  File "/usr/local/lib/python2.7/dist-packages/allure_commons/_allure.py", line 169, in impl
    return func(*a, **kw)
  File "/root/mars/workspace/sonic-mgmt/tests/vxlan/test_vxlan_ecmp.py", line 215, in get_ethernet_to_neighbors
    ret_list.append(intf)
KeyError: 'address'
`

Reply:
> on T1-lag, and it was working.

I agree, it works, because for LAG topo only `get_portchannels_to_neighbors` invoked not `get_ethernet_to_neighbors`

Reply:
I will revert this particular change. Pls let me know if it works after reverting it. If I can have more time, I will test it on T1 topo, if and when I get access to it.